### DISCUSSION TITLE
Uploading file using pre-signed upload link

### DISCUSSION THREAD

**Discussion by dmitrykaigorodov** [pocketbase/pocketbase#916] [Q&A]
When Pocketbase is configured to use S3, does it upload files directly to S3? 

Does it use pre-signed upload links?

Can I upload files to S3 without my files ever touching the Pocketbase web-server?

Thank you!

**Comment by ganigeorgiev** [comment]
> When Pocketbase is configured to use S3, does it upload files directly to S3?

Yes, via the S3 API.

---

> Does it use pre-signed upload links?

No, PocketBase uses the S3 API and support any S3 compatible service not just AWS. The authentication is done via the user provided access and secret keys (you could check https://github.com/pocketbase/pocketbase/blob/rc/tools/filesystem/filesystem.go#L34-L65).

---

> Can I upload files to S3 without my files ever touching the Pocketbase web-server?

No, you'll have to use the PocketBase record create API since you need also the filename of the uploaded file so that it can be linked to your record.

_I have never used the aws presigned upload links, but reading the docs, you can achieve similar (and probably easier) behavior by using directly the PocketBase REST create record api in combination with suitable create api rule for your collection._

**Comment by dmitrykaigorodov** [comment]
Just to give it more context, here what I mean when say pre-signed upload link: 
https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html
